src/runtime/pkg/govmm/qemu/qemu.go

```diff
@@ -3058,15 +3071,21 @@ func LaunchCustomQemu(ctx context.Context, path string, params []string, fds []*
 
 	cmd.SysProcAttr = attr
 
-	var stderr bytes.Buffer
-	cmd.Stderr = &stderr
+	stderrIn, _ := cmd.StderrPipe()
 	logger.Infof("launching %s with: %v", path, params)
 
-	err := cmd.Run()
+	err := cmd.Start()
 	if err != nil {
 		logger.Errorf("Unable to launch %s: %v", path, err)
+		stderr := new(bytes.Buffer)
+		io.Copy(stderr, stderrIn)
 		errStr = stderr.String()
 		logger.Errorf("%s", errStr)
+		return nil, errStr, err
+	}
+
+	if logFunc != nil {
+		go logFunc(cmd.Process.Pid, stderrIn)
```

Comment:
This go routine needs to be running before line 3084, right? Otherwise the go routine might never start. Maybe move it to line 3079 and synchronize it with the main thread.

Reply:
> This go routine needs to be running before line 3084, right? Otherwise the go routine might never start. Maybe move it to line 3079 and synchronize it with the main thread.

Line 3084 is on the error path when QEMU couldn't be started. We don't need a goroutine in this case. Just collect the errors from `cmd.Start()` ... but thanks to your remark I know realize there's more to it : if QEMU didn't start then we don't need to collect stderr. I'll simplify the error path.